it/v2/connectors/PersistenceConnectorSpec.scala

```diff
@@ -921,6 +921,176 @@ class PersistenceConnectorSpec
     }
   }
 
+  "GET /traders/:eori/movements/arrivals/:arrivalId/messages" - {
+
+    implicit val hc = HeaderCarrier()
+
+    def targetUrl(eoriNumber: EORINumber, arrivalId: MovementId) =
+      s"/transit-movements/traders/${eoriNumber.value}/movements/arrivals/${arrivalId.value}/messages/"
+
+    def targetUrlWithTime(eoriNumber: EORINumber, arrivalId: MovementId, receivedSince: OffsetDateTime) =
+      s"/transit-movements/traders/${eoriNumber.value}/movements/arrivals/${arrivalId.value}/messages/?receivedSince=${DateTimeFormatter.ISO_OFFSET_DATE_TIME
+        .format(receivedSince)}"
+
+    "on successful return of message IDs when no filtering is applied, return a success" in forAll(
+      arbitrary[EORINumber],
+      arbitrary[MovementId],
+      Gen.nonEmptyListOf(arbitrary[MessageSummary])
+    ) {
+      (eori, arrivalId, messageResponse) =>
+        server.stubFor(
+          get(
+            urlEqualTo(targetUrl(eori, arrivalId))
+          ).willReturn(
+            aResponse()
+              .withStatus(OK)
+              .withBody(
+                Json.toJson(messageResponse).toString()
+              )
+          )
+        )
+        val result = persistenceConnector.getArrivalMessageIds(eori, arrivalId, None)
+        whenReady(result) {
+          _ mustBe messageResponse
+        }
+    }
+
+    "on successful return of arrival message IDs when filtering by received date is applied, return a success" in forAll(
+      arbitrary[EORINumber],
+      arbitrary[MovementId],
+      OffsetDateTime.now(ZoneOffset.UTC),
```

Comment:
You're probably better off using an arbitrary `OffsetDateTime` instead rather than `now`.

Reply:
when I am using arbitrary OffsetDateTime it is failing with **request was not matched** and I can see some encoded value for time. Not sure how to use arbitrary here

Reply:
It makes sense.

```
GET                                                        | GET
/transit-movements/traders//movements/arrivals//messages/  | /transit-movements/traders//movements/arrivals//messages/<<<<< URL does not match
?receivedSince=+60085274-09-20T10:23:43.300069439-16:10:2  | ?receivedSince=%2B60085274-09-20T10:23:43.300069439-16:10
2                                                          | :22
```

`+` is becoming `%2B` when the request is made, which is entirely legitimate because URLs _should_ be URL encoded. However, and this is a problem with Wiremock I guess, or quirk at least, you're not looking for the URL encoded URL.

The problem with the time you gave is that you weren't potentially testing the + sign against the URLs. Now, thankfully, the code is doing the right thing in URL encoding the URL (see https://developer.service.hmrc.gov.uk/api-documentation/docs/api/service/common-transit-convention-traders/2.0#percentencoding-of-parameters-in-request-urls) -- it is your test that is listening on the wrong URL.

Fix how `targetUrlWithTime` is encoding the URL (do something like `${URLEncoder.encode(DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(receivedSince), StandardCharsets.UTF_8.name())}`).